builder_union.go

```diff
@@ -11,24 +11,39 @@ import (
 )
 
 func (b *Builder) unionWriteTo(w Writer) error {
-	for idx, v := range b.unions {
-		if v.builder.optype != selectType {
+	for idx, u := range b.unions {
+		current := u.builder
+		if current.optype != selectType {
 			return errors.New("UNION is only allowed among SELECT operations")
 		}
+
 		if len(b.unions) == 1 {
-			if err := v.builder.selectWriteTo(w); err != nil {
+			if err := current.selectWriteTo(w); err != nil {
 				return err
 			}
 		} else {
 			if idx != 0 {
-				fmt.Fprint(w, fmt.Sprintf(" UNION %v ", strings.ToUpper(v.unionType)))
+				fmt.Fprint(w, fmt.Sprintf(" UNION %v ", strings.ToUpper(u.unionType)))
 			}
 			fmt.Fprint(w, "(")
-			if err := v.builder.selectWriteTo(w); err != nil {
-				return err
+
+			if current.limitation != nil {
+				tw := NewWriter()
```

Comment:
It seems just use `current.selectWriteTo(w)` is OK?

Reply:
Yeah, actually it only works when performing oracle & mssql LIMIT op. I‘ll provide an optimized solution

Reply:
Yeah, actually it only works when performing oracle & mssql LIMIT op. I‘ll provide an optimized solution

Reply:
As the first one😀